<?xml version="1.0"?>

<launch>
    <arg name="rviz" default="true" />
    <arg name="gui" default="true"/>

    <arg name="robot_ip" default="192.168.1.164"/>
    
    <include file="$(find ur_modern_driver)/launch/ur10_bringup.launch">
        <arg name="robot_ip" value="$(arg robot_ip)"/>
    </include>

    <arg name="camera_name" default="phobos_nuclear"/>
    
    <!--<arg name="cam_robot_tf" default="-0.0475 0.0475 0 0, 0.0766334, -0.2682168, -0.9603057 "/>-->
    <arg name="cam_robot_tf" default="0 0 0 0 0 1 0"/>
    <!--<arg name="aruco_robot_base_tf" default="0.5 0.024752 -0.20641 0 0 0 -1"/>-->

    <node name="camera_robot_publisher" pkg="tf" type="static_transform_publisher" args="$(arg cam_robot_tf) tool0 $(arg camera_name)_base_link 100" />

    <!--<node name="camera_robot_publisher" pkg="tf" type="static_transform_publisher" args="$(arg cam_robot_tf) map tool0 100" />-->

    <!--<node name="i3dr_hand_eye" pkg="i3dr_ur10" type="i3dr_hand_eye" output="screen">
        <param name="camera_namespace" type="string" value="$(arg camera_name)"/>
        <param name="camera_base_frame" type="string" value="$(arg camera_name)_base_link"/>
        <param name="camera_frame" type="string" value="$(arg camera_name)_depth_optical_frame"/>
        <param name="world_frame" type="string" value="world"/>
        <param name="arm_tooltip_frame" type="string" value="tool0"/>
        <param name="arm_base_frame" type="string" value="base_link"/>
        <param name="publish_aruco_tf" type="bool" value="true"/>
        <param name="image_flipped" type="bool" value="false"/>

        <param name="camera_mounted_on_robot" type="bool" value="false"/>
    </node>-->

   <node name="i3dr_ur10_scan" pkg="i3dr_ur10" type="i3dr_ur10_scan2.py" output="screen">
        <param name="rtabmap_en" type="bool" value="false"/>
        <param name="ur10_move_time" type="int" value="3"/>
        <param name="ur10_pose_pause" type="int" value="0"/>
        <param name="rtabmap_namespace" type="string" value="$(arg camera_name)"/>
    </node>

    <node if="$(arg gui)" name="i3dr_ur10_scan_gui" pkg="i3dr_ur10" type="i3dr_ur10_scan_gui.py" output="screen">
        <remap from="left/image_raw" to="$(arg camera_name)/left/image_raw"/>
        <remap from="right/image_raw" to="$(arg camera_name)/right/image_raw"/>
        <remap from="disparity" to="$(arg camera_name)/disparity"/>
    </node>
 
    <!-- DISPLAY -->
    <!-- run rviz with config file -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rivz" 
        args="-d $(find i3dr_ur10)/rviz/ur10_sbri.rviz">
        <!--<remap from="rtabmap/get_map_data" to="$(arg camera_name)/get_map_data"/>-->
    </node>
</launch>